libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -17,7 +22,28 @@ ReturnCode CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element
         return ReturnCode::FAILURE;
     }
 
-    result.first = settingKey;
-    result.second = settingValue;
+    log_debug() << "key : " << settingKey << " value : " << settingValue;
+
+    if ("memory.limit_in_bytes" == settingKey) {
+        if (m_settings.count(settingKey)) {
+            // if the new value is lexicographical greater than old one set is as the value
```

Comment:
I just change it to integer comparision after deciding to apply only one setting